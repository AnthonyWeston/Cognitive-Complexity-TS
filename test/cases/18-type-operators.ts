type A = true;

type B = true | true | true; // series of 1 operator

type C = true | true & false; // series of 2 operators

type D = true | true & false & false | true | true; // series of 3 operators

type E = true & true | false | false & true & true; // series of 3 operators

type F = (true | true) | true;

type G = true | (true & false);
