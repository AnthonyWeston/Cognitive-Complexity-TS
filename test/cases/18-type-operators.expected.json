{
    "18-type-operators.ts": {
        "score": 13,
        "inner": [
            {
                "name": "A",
                "score": 0,
                "line": 1,
                "column": 1
            },
            {
                "name": "B",
                "score": 1,
                "line": 3,
                "column": 1
            },
            {
                "name": "C",
                "score": 2,
                "line": 5,
                "column": 1
            },
            {
                "name": "D",
                "score": 3,
                "line": 7,
                "column": 1
            },
            {
                "name": "E",
                "score": 3,
                "line": 9,
                "column": 1
            },
            {
                "name": "F",
                "score": 2,
                "line": 11,
                "column": 1
            },
            {
                "name": "G",
                "score": 2,
                "line": 13,
                "column": 1
            }
        ]
    }
}